pkg/resource/api.go

```diff
@@ -154,6 +154,10 @@ func NewAPIManagedBinder(c client.Client) *APIManagedBinder {
 // Bind the supplied resource to the supplied claim.
 func (a *APIManagedBinder) Bind(ctx context.Context, cm Claim, mg Managed) error {
 	cm.SetBindingPhase(v1alpha1.BindingPhaseBound)
+	// Propagate back the final name of the external resource to the claim.
+	if meta.GetExternalName(mg) != "" {
+		meta.SetExternalName(cm, meta.GetExternalName(mg))
```

Comment:
I believe setting the claim's external name will require us to call `client.Update(cm)`. It looks like we currently only call `client.Status().Update(cm)`. My preference would be to call `client.Update` from inside the `APIManagedBinder` and `APIManagedStatusBinder`.

Reply:
Yes, I realized that it didn't propagate back while testing new CloudSQL controller. Thanks for spotting!